### ISSUE TITLE
Some errors only show when the shrinkwrap is present

### ISSUE THREAD

**Issue by majgis** [nodesecurity/nsp#119]
I reported a similar error in the past and it was fixed.  This is either  a regression or a different issue.

```
$ npm init -f
npm WARN using --force I sure hope you know what you are doing.
Wrote to /Users/user/projects/moonchild/tmp/package.json:

{
  "name": "tmp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}



$ npm i express@4.13.3 --save
npm WARN package.json tmp@1.0.0 No description
npm WARN package.json tmp@1.0.0 No repository field.
npm WARN package.json tmp@1.0.0 No README data
express@4.13.3 node_modules/express
├── escape-html@1.0.2
├── merge-descriptors@1.0.0
├── array-flatten@1.1.1
├── cookie@0.1.3
├── utils-merge@1.0.0
├── cookie-signature@1.0.6
├── methods@1.1.2
├── fresh@0.3.0
├── range-parser@1.0.3
├── vary@1.0.1
├── path-to-regexp@0.1.7
├── parseurl@1.3.1
├── etag@1.7.0
├── content-type@1.0.2
├── content-disposition@0.5.0
├── depd@1.0.1
├── qs@4.0.0
├── on-finished@2.3.0 (ee-first@1.1.1)
├── proxy-addr@1.0.10 (forwarded@0.1.0, ipaddr.js@1.0.5)
├── debug@2.2.0 (ms@0.7.1)
├── finalhandler@0.4.0 (unpipe@1.0.0)
├── accepts@1.2.13 (negotiator@0.5.3, mime-types@2.1.11)
├── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)
├── send@0.13.0 (destroy@1.0.3, statuses@1.2.1, ms@0.7.1, mime@1.3.4, http-errors@1.3.1)
└── serve-static@1.10.3 (escape-html@1.0.3, send@0.13.2)

$ nsp -v
2.4.0

$ nsp check
(+) No known vulnerabilities found

$ npm shrinkwrap
wrote npm-shrinkwrap.json

$ nsp check
(+) 1 vulnerabilities found
┌───────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│               │ Regular Expression Denial of Service                                                                                                                │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Name          │ negotiator                                                                                                                                          │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Installed     │ 0.5.3                                                                                                                                               │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <= 0.6.0                                                                                                                                            │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Patched       │ >= 0.6.1                                                                                                                                            │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Path          │ tmp@1.0.0 > express@4.13.3 > accepts@1.2.13 > negotiator@0.5.3                                                                                      │
├───────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/106                                                                                                              │
└───────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

```

**Comment by nlf** [comment]
i'll take a look at this and see if i can figure out what's going on

**Comment by nlf** [comment]
ok, so i took a look at this, and the problem is that when you `npm i express@4.13.3 --save` it installs express@4.13.3, however it adds `"express": "^4.13.3"` to the dependencies in your package.json. since the nsp client tests against the newest matching semver available (not what's installed locally) it ends up testing against 4.14.0 where this bug has been fixed.

the reason it shows up when you shrinkwrap is because your shrinkwrap is generated based on what's locally installed, in this case version 4.13.3, and we respect the versions there.

tl;dr your package.json is asking for a version range that includes the latest (fixed) express release

**Comment by nlf** [comment]
no problem, thanks for bringing it up!
